tests/unit/models/command-test.js

```diff
@@ -558,6 +561,96 @@ describe('models/command.js', function() {
     });
   });
 
+  describe('runTask', function() {
+    let command;
+
+    class AsyncTask extends Task {
+      run(options) {
+        return new Promise(function(resolve) {
+          setTimeout(() => resolve(options), 50);
+        });
+      }
+    }
+
+    class SyncTask extends Task {
+      run(options) { return options; }
+    }
+
+    class FailingTask extends Task {
+      run(options) { throw new Error('I was born to fail'); }
+    }
+
+    beforeEach(function() {
+      // this should be changed to new Command(), but needs more mocking
+      command = new ServeCommand(Object.assign({}, options, {
+        tasks: {
+          Async: AsyncTask,
+          Sync: SyncTask,
+          Failing: SyncTask,
+        },
+      }));
+    });
+
+    it('always handles task as a promise', function() {
+      return command.runTask('Sync', { param: 'value' }).then(result => {
+        expect(result).to.eql({
+          param: 'value',
+        });
+      });
+    });
+
+    it('command environment should be shared with a task', function() {
+      let taskRun = command.runTask('Async', { param: 'value' });
+
+      expect(command._currentTask.ui).to.eql(command.ui);
+      expect(command._currentTask.analytics).to.eql(command.analytics);
+      expect(command._currentTask.project).to.eql(command.project);
+
+      return taskRun;
+    });
+
+    it('_currentTask should store a reference to the current task', function() {
+      expect(command._currentTask).to.be.undefined;
+      let taskRun = command.runTask('Sync', { param: 'value' }).then(result => {
+        expect(command._currentTask).to.be.undefined;
+      });
+      expect(command._currentTask).to.be.an.instanceof(SyncTask);
+
+      return taskRun;
+    });
+
+    it('_currentTask should cleanup current task on fail', function() {
+      return command.runTask('Failing', { param: 'value' })
+        .then(() => expect(false, 'should not be resolved').to.equal(true))
+        .catch(result => expect(command._currentTask).to.be.undefined);
+    });
+
+    it('throws on attempt to launch concurrent tasks', function() {
+      let asyncTaskRun, syncTaskRun;
```

Comment:
oops...

I've decided to handle all the tasks as promises to make tests bullet proof. I'm going to use this variable now.